Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp

```diff
@@ -201,13 +202,32 @@ namespace AtomToolsFramework
 
     bool ModularViewportCameraControllerInstance::HandleInputChannelEvent(const AzFramework::ViewportControllerInputEvent& event)
     {
+        auto modifierKeyStates = AzFramework::ModifierKeyStates{};
+        const auto* inputDevice =
+            AzFramework::InputDeviceRequests::FindInputDevice(AzToolsFramework::GetSyntheticKeyboardDeviceId(event.m_viewportId));
+
+        // grab keyboard channel (not important which) and check the modifier state (modifier state is shared for all keyboard channels)
+        if (const auto it = inputDevice->GetInputChannelsById().find(AzFramework::InputDeviceKeyboard::Key::Alphanumeric0);
+            it != inputDevice->GetInputChannelsById().end())
+        {
+            modifierKeyStates = [customData = it->second->GetCustomData<AzFramework::ModifierKeyStates>()]
+            {
+                if (customData)
+                {
+                    return *customData;
+                }
+
+                return AzFramework::ModifierKeyStates();
+            }();
```

Comment:
Creating a lambda to immediately call it seems unusual, is it necessary?

Reply:
I like doing this so we can keep the local variable `const` - there's actually a nice bit of guidance in the C++ Core Guidelines which mention doing this as good practice - https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-lambda-init.

I could name the lambda first to aid with readability if you really don't like it but this is a pattern that already crops up a fair bit in the O3DE code base so it's not exactly without precedent 😛

Reply:
modifierKeyStates isn't const though?

Reply:
Ah good spot, this likely went through a few refactorings and I missed that. I'm away from my machine this morning (working from a Mac in the office) but will look back that this later today and see if I can simplify it :+1:

Reply:
FYI I did actually refactor this after you mentioned it and I think it's a lot clearer now :+1: